src/main/java/com/microsoft/graph/Requests/BaseGraphRequestAdapter.java

```diff
@@ -0,0 +1,56 @@
+package com.microsoft.graph.Requests;
+
+import com.microsoft.kiota.authentication.AuthenticationProvider;
+import com.microsoft.kiota.http.OkHttpRequestAdapter;
+import com.microsoft.kiota.serialization.ParseNodeFactory;
+import com.microsoft.kiota.serialization.SerializationWriterFactory;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import okhttp3.OkHttpClient;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.HashMap;
+
+@SuppressFBWarnings
+public class BaseGraphRequestAdapter extends OkHttpRequestAdapter {
+
+    enum Clouds {
+        GLOBAL_CLOUD,
+        USGOV_CLOUD,
+        CHINA_CLOUD,
+        GERMANY_CLOUD,
+        USGOV_DOD_CLOUD
+    }
+
+    private static final HashMap<Clouds, String> cloudList = new HashMap<Clouds, String>() {{
+        put( Clouds.GLOBAL_CLOUD, "https://graph.microsoft.com" );
+        put( Clouds.USGOV_CLOUD, "https://graph.microsoft.us");
+        put( Clouds.CHINA_CLOUD, "https://microsoftgraph.chinacloudapi.cn");
+        put( Clouds.GERMANY_CLOUD, "https://graph.microsoft.de");
+        put( Clouds.USGOV_DOD_CLOUD, "https://dod-graph.microsoft.us");
+    }};
+
+    public BaseGraphRequestAdapter(@Nonnull final AuthenticationProvider authenticationProvider, @Nullable final ParseNodeFactory parseNodeFactory, @Nullable final SerializationWriterFactory serializationWriterFactory, @Nullable final OkHttpClient client, @Nullable GraphClientOptions graphClientOptions, @Nullable String baseUrl) {
```

Comment:
we probably should also offer overload constructors (emulating optional parameters in effects)